drv/transceivers-server/src/main.rs

```diff
@@ -503,11 +536,11 @@ impl NotificationHandler for ServerImpl {
                 Err(_) => ModulesStatus::new_zeroed(),
             };
 
-            if status.present != self.modules_present {
-                self.led_update(status.present);
+            if !status.modprsl != self.modules_present {
+                self.led_update(!status.modprsl);
 
-                self.modules_present = status.present;
-                ringbuf_entry!(Trace::ModulePresenceUpdate(status.present));
+                self.modules_present = !status.modprsl;
+                ringbuf_entry!(Trace::ModulePresenceUpdate(!status.modprsl));
```

Comment:
How about assigning `!status.modprsl` to a local variable here, e.g.
```rust
let modules_present = !status.modprsl;
 if modules_present != self.modules_present {
    self.led_update(modules_present);

    self.modules_present = modules_present;
    ringbuf_entry!(Trace::ModulePresenceUpdate(modules_present));
 }
```

Reply:
:+1: